client/src/sections/resources/IndustrialFireExtinguisher.tsx

```diff
@@ -0,0 +1,97 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import useGet from "../../hooks/useGet";
+import { Step, useQuestStep } from "../../hooks/useQuest";
+import { $item } from "../../util/makeValue";
+
+const IndustrialFireExtinguisher = () => {
+    const foam = useGet("_fireExtinguisherCharge");
+
+    const knobUnfinished = useQuestStep("questL05Goblin") < Step.FINISHED;
+    const knobUnused = !useGet("fireExtinguisherHaremUsed");
+    const hasPants = useHave($item`Knob Goblin harem pants`);
+    const hasVeil = useHave($item`Knob Goblin harem veil`);
+    const canVisitTheThrone = hasPants && hasVeil;
+    const showKnob =
+        useQuestStep("questL05Goblin") > Step.UNSTARTED &&
+        !canVisitTheThrone &&
+        knobUnused &&
+        knobUnfinished;
+
+    const bridgeUnfinished = useQuestStep("questL09Topping") < 1;
+    const blechUnused = !useGet("fireExtinguisherChasmUsed");
+    const showBlech =
+        useQuestStep("questL09Topping") > Step.UNSTARTED &&
+        bridgeUnfinished &&
+        blechUnused;
+
+    const batwallsUnfinished = useQuestStep("questL04Bat") < 3;
+    const batUnused = !useGet("fireExtinguisherBatHoleUsed");
+    const showBat =
+        useQuestStep("questL04Bat") > Step.UNSTARTED &&
+        batwallsUnfinished &&
+        batUnused;
+
+    const bonerUnready = [
+        useGet("cyrptAlcoveEvilness"),
+        useGet("cyrptCrannyEvilness"),
+        useGet("cyrptNicheEvilness"),
+        useGet("cyrptNookEvilness"),
+    ].some((evilness) => evilness > 25);
+    const cyrptUnused = !useGet("fireExtinguisherCyrptUsed");
+    const showCyrpt =
+        useQuestStep("questL07Cyrptic") > Step.UNSTARTED &&
+        bonerUnready &&
+        cyrptUnused;
+
+    const desertIncomplete = useGet("desertExploration") < 100;
```

Comment:
do we want to add conditionals on desert exploration <100 but they still have exploration helpers to use? i lean no because there are complexities about quest access for things like can of black paint, so it's a bit complex for the minimum viable product, but figured i'd raise it as something we might want to comment as a future improvement at least.